RadarSDK/RadarAddress.m

```diff
@@ -0,0 +1,187 @@
+//
+//  RadarAddress.m
+//  RadarSDK
+//
+//  Copyright © 2019 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarAddress+Internal.h"
+#import "RadarCoordinate+Internal.h"
+
+@implementation RadarAddress
+
++ (NSArray<RadarAddress *> * _Nullable)addressesFromObject:(id _Nonnull)object {
+    if (!object || ![object isKindOfClass:[NSArray class]]) {
+        return nil;
+    }
+
+    NSArray *addressesArr = (NSArray *)object;
+    NSMutableArray<RadarAddress *> *mutableAddresses = [NSMutableArray<RadarAddress *> new];
+
+    for (id addressObj in addressesArr) {
+        RadarAddress *address = [[RadarAddress alloc] initWithObject:addressObj];
+        if (!address) {
+            return nil;
+        }
+        [mutableAddresses addObject:address];
+    }
+
+    return mutableAddresses;
+}
+
+- (instancetype _Nullable)initWithObject:(id _Nonnull)object {
+    if (!object || ![object isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    NSDictionary *addressDict = (NSDictionary *)object;
+
+    NSNumber *latitude;
+    NSNumber *longitude;
+    RadarCoordinate *coordinate;
+
+    NSString *formattedAddress;
+    NSString *country;
+    NSString *countryCode;
+    NSString *countryFlag;
+    NSString *state;
+    NSString *stateCode;
+    NSString *postalCode;
+    NSString *city;
+    NSString *borough;
+    NSString *county;
+    NSString *neighborhood;
+    NSString *number;
+
+    // TODO (jsani): is this a sensible default for 'confidence', or should we have another enum entry?
+    RadarAddressConfidence confidence = RadarAddressConfidenceFallback;
```

Comment:
For event confidences, we default to `"none"` (or `0`) - we probably want to do the same thing here, since `fallback` means something more specific.

https://github.com/radarlabs/radar-sdk-ios/blob/v3/RadarSDK/RadarEvent.m#L77